mypy/errorcodes.py

```diff
@@ -0,0 +1,50 @@
+"""Classification of possible errors mypy can detect.
+
+These can be used for filtering specific errors.
+"""
+
+from typing import List
+from typing_extensions import Final
+
+
+# All created error codes are implicitly stored in this list.
+all_error_codes = []  # type: List[ErrorCode]
+
+
+class ErrorCode:
+    def __init__(self, code: str, description: str, category: str) -> None:
+        self.code = code
+        self.description = description
+        self.category = category
```

Comment:
What are the plans for `category`?

Reply:
My idea is to use it for grouping the available error codes in the output from `--show-error-codes`, once that's implemented. This way error codes defined by plugins, for example, can be shown in a separate section.

Reply:
I actually meant a separate command such as `--list-error-codes` that shows all the available error codes.